### DISCUSSION TITLE
Migration to V1

### DISCUSSION THREAD

**Discussion by huwmanatee** [SixLabors/ImageSharp#1326] [Q&A] [answered]
I used to use SixLabors.ImageSharp.Processing.Processors.Convolution.SobelProcessor in 1.0.0-beta0008, it seems to have been removed or renamed in V1.

For broader context, I am using the Sobel Processor to find a bounding box for forms that have been scanned in to position the content.

Here is some F# code:
```
let detectImageBounds (image:Image<'TPixel>, sourceRectangle) =
            let sobelProcessor = SobelProcessor(true)
            let sobelProcessor = sobelProcessor.CreatePixelSpecificProcessor(image, sourceRectangle)
            sobelProcessor.Apply()

            let getFilteredBoundingRectangle (imageFrame: ImageFrame<'TPixel>, componentValue:float32 ) =
                let t = Type.GetType("SixLabors.ImageSharp.ImageMaths, SixLabors.ImageSharp", false)

                let m =
                    t.GetMethods()
                    |> Seq.tryFind(fun m -> m.Name.StartsWith("GetFilteredBoundingRectangle"))
                    |> Option.toObj

                let gm = m.MakeGenericMethod(typeof<'TPixel>)
                gm.Invoke(null, [| box imageFrame; box componentValue; box 2 |]) :?> SixLabors.ImageSharp.Rectangle

            getFilteredBoundingRectangle (image.Frames.RootFrame, 0.0f)
```

I am using some internal stuff to achieve this, specifically I use reflection to get the bounding rectangle, perhaps there is an easier way?

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah `GetFilteredBoundingRectangle` is internal, we've no plans currently to expose the method as I would like to refactor it first  so I would copy the code rather than use reflection as I'm not even happy with the name. 

Regarding `SobelProcessor` yeah that was changed as part of https://github.com/SixLabors/ImageSharp/pull/1299 where we refactored the edge detection API to allow passing of custom kernels. `EdgeDetector2DProcessor(KnownEdgeDetectorKernels.Sobel, true)` performs the equivalent operation.

The equivalent code that we use to calculate the bounding box for entropy cropping can be found in the source here.

https://github.com/SixLabors/ImageSharp/blob/8e2792e3ab96dd69d8e39255e83afdee493b1b26/src/ImageSharp/Processing/Processors/Transforms/EntropyCropProcessor%7BTPixel%7D.cs#L44-L51

**Reply by huwmanatee** [reply]
You have just reminded me of how I came up with the code in the first place, by looking at what the crop processor does.

Thank you very much.

**Reply by JimBobSquarePants** [reply]
Happy to help! 😄
